src/com/hms_networks/americas/sc/realtimedata/RealTimeTagDataPointManager.java

```diff
@@ -0,0 +1,82 @@
+package com.hms_networks.americas.sc.realtimedata;
+
+import java.util.ArrayList;
+
+import com.ewon.ewonitf.EWException;
+import com.ewon.ewonitf.TagControl;
+import com.hms_networks.americas.sc.datapoint.DataPoint;
+import com.hms_networks.americas.sc.logging.Logger;
+
+/**
+ * This class will hold a list of data points for each tag. One instance of the class is made per
+ * tag.
+ */
+public class RealTimeTagDataPointManager {
+  /** The name of the tag */
+  public String tagName;
+
+  /** A list of data points retrieved for the tag */
+  private ArrayList dataPoints;
+
+  /** The last data value retrieved for this tag */
+  private DataPoint lastDataPoint;
+
+  /** hold a tag control object to use when retrieving data points */
+  public TagControl tagControl;
+
+  /**
+   * Gets the tag's name
+   *
+   * @return The tag name
+   */
+  public String getTagName() {
+    return tagName;
+  }
+
+  /**
+   * Constructor
+   *
+   * @param tagName The name of the tag that data points are coming from
+   */
+  public RealTimeTagDataPointManager(String tagName) {
+    this.tagName = tagName;
+    this.lastDataPoint = null;
+    this.dataPoints = new ArrayList();
+
+    try {

```

Comment:
for the most part, try/catches shouldn't exist in the libraries -- unless necessary. The method should throw it, that way each application can handle the error in whatever way is desired. Try/catches in the library are not visible from the application really.